src/syscall.cr

```diff
@@ -4,23 +4,24 @@
 # For more details about Linux system calls, refer to
 # [https://man7.org/linux/man-pages/man2/syscalls.2.html](https://man7.org/linux/man-pages/man2/syscalls.2.html).
 #
-# To define system calls reopen this module and use the `def_syscall` macro. Pass in the system call name,
-# the return type and its arguments. For example:
-#
-#   module Syscall
-#     def_syscall pipe2, Int32, pipefd : Int32[2]*, flags : Int32
-#     def_syscall write, Int32, fd : Int32, buf : UInt8*, count : LibC::SizeT
-#     def_syscall read, Int32, fd : Int32, buf : UInt8*, count : LibC::SizeT
-#     def_syscall close, Int32, fd : Int32
-#   end
-#
-# They can then be called as methods from the `Syscall` module:
-#
-#   Syscall.write(1, "Hello!\n".to_unsafe, 7)
-#
+
 @[Experimental]
 module Syscall
-  private macro def_syscall(name, return_type, *args)
+  # To define system calls open a module and use the `def_syscall` macro. Pass in the system call name,
+  # the return type and its arguments. For example:
+  #
+  #   module YourSyscalls
+  #     Syscall.def_syscall pipe2, Int32, pipefd : Int32[2]*, flags : Int32
+  #     Syscall.def_syscall write, Int32, fd : Int32, buf : UInt8*, count : LibC::SizeT
+  #     Syscall.def_syscall read, Int32, fd : Int32, buf : UInt8*, count : LibC::SizeT
+  #     Syscall.def_syscall close, Int32, fd : Int32
+  #   end
+  #
+  # They can then be called as methods from the `Syscall` module:
+  #
+  #   YourSyscalls.write(1, "Hello!\n".to_unsafe, 7)
```

Comment:
The source code blocks need to be indented by four spaces, or delimited by triple backticks. I prefer backticks because that's the more common style in stdlib.

Reply:
Fixed.